### DISCUSSION TITLE
Empty Expression

### DISCUSSION THREAD

**Discussion by lglaz** [datalust/superpower#138] [Q&A]
I'm trying to write a parser for a filter string which can be translated into SQL WHERE clause or some client-side table filter. How to write a parser where an empty expression (meaning no filtering) is a valid part of the grammar?

**Comment by nblumhardt** [comment]
Hi Łukasz,

If you mean that the whole input being empty implies `true`, then the top level parser might look something like:

```csharp
static readonly TokenListParser<XToken, XExpression> Filter =
    Expression.OptionalOrDefault((XExpression)new XConstantExpression(true)).AtEnd();
```

(Assuming your language is called `X` :-))

Where I've encountered this, though, I tend to just check the input first with `string.IsNullOrWhiteSpace()` and skip parsing altogether.

HTH!

**Reply by lglaz** [reply]
Thanks for the answer and the additional tip. I'll probably use the latter :)
